bazel/repository_locations.bzl

```diff
@@ -322,26 +322,26 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_desc = "Data Collect Protocols of Apache SkyWalking",
         project_url = "https://github.com/apache/skywalking-data-collect-protocol",
         name = "skywalking_data_collect_protocol",
-        sha256 = "fa9ac679624217f30b6e8d5c450365386c610e2d08188a20f0340c3b14401833",
-        urls = ["https://github.com/apache/skywalking-data-collect-protocol/archive/v8.3.0.zip"],
-        strip_prefix = "skywalking-data-collect-protocol-8.3.0",
-        version = "8.3.0",
+        sha256 = "edfa970394511213eacc8055b4c13e4e9773e9196122a49e0db68f6162f67dff",
+        urls = ["https://github.com/apache/skywalking-data-collect-protocol/archive/v{version}.tar.gz"],
+        strip_prefix = "skywalking-data-collect-protocol-{version}",
+        version = "8.4.0",
         use_category = ["observability_ext"],
         extensions = ["envoy.tracers.skywalking"],
-        release_date = "2020-11-20",
+        release_date = "2021-01-20",
```

Comment:
The UTC release date for this release is `2021-02-01`. I suspect this value works as this is the date of the last commit for the release.